keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/ReshapeVertex.java

```diff
@@ -0,0 +1,53 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
+
+import io.improbable.keanu.tensor.TensorShape;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
+
+import java.util.Map;
+
+public class ReshapeVertex extends DoubleUnaryOpVertex {
+
+    private int[] proposedShape;
+
+    public ReshapeVertex(DoubleVertex inputVertex, int... proposedShape) {
+        super(inputVertex.getShape(), inputVertex);
```

Comment:
The output shape should be the proposedShape not the same shape as the inputVertex.

Reply:
Good spot, changed